crates/wasm-smith/src/core.rs

```diff
@@ -918,36 +918,48 @@ impl Module {
         );
 
         let mut export_names = HashSet::new();
-        arbitrary_loop(
-            u,
-            self.config.min_exports(),
-            self.config.max_exports(),
-            |u| {
-                // Remove all candidates for export whose type size exceeds our
-                // remaining budget for type size. Then also remove any classes
-                // of exports which no longer have any candidates.
-                //
-                // If there's nothing remaining after this, then we're done.
-                let max_size = self.config.max_type_size() - self.type_size;
-                for list in choices.iter_mut() {
-                    list.retain(|(kind, idx)| self.type_of(*kind, *idx).size() + 1 < max_size);
-                }
-                choices.retain(|list| !list.is_empty());
-                if choices.is_empty() {
-                    return Ok(false);
+        if self.config.export_everything() {
+            for choices_by_kind in choices {
+                for (kind, idx) in choices_by_kind {
+                    let name = unique_string(1_000, &mut export_names, u)?;
+                    let ty = self.type_of(kind, idx);
+                    self.type_size += 1 + ty.size();
+                    self.exports.push((name, kind, idx));
                 }
+            }
+            Ok(())
```

Comment:
Can this be `return Ok(())` and then we can remove the `else` and the extra level of indentation below?